runtime/lua/vim/lsp/util.lua

```diff
@@ -2139,7 +2139,7 @@ end
 ---
 ---@param settings table language server settings
 ---@param section  string indicating the field of the settings table
----@return table|string The value of settings accessed via section
+---@return table|string|vim.NIL The value of settings accessed via section. `vim.NIL` if not found.
 function M.lookup_section(settings, section)
```

Comment:
In retrospect, it wasn't a great choice this (utility) function is returning `vim.NIL` instead of `nil`. Should we deprecate this function and mark it as private in the (near) future?